File: oner-python/src/app/infrastructure/models/Inventory/pharmacy_item.py
from sqlalchemy.sql.schema import ForeignKey
from sqlalchemy.sql.sqltypes import Integer, String
from sqlalchemy.orm import relationship
from sqlalchemy import Column
from app.infrastructure.models.baseModel import BaseModel

class PharmacyItem(BaseModel):
    __tablename__='pharmacy_item'

    pharmacy_category_id=Column(Integer,ForeignKey('pharmacy_category.id'))
    category=relationship('PharmacyCategory')
    universal_product_code=Column(String)
    brand_name=Column(String)
    generic_name=Column(String)
    form=Column(String)
    strength=Column(String)
    packaging=Column(String)
    uom=Column(String)
    service_item=relationship('ServiceItem',primaryjoin="and_(remote(ServiceItem.relation_id)==foreign(PharmacyItem.id),ServiceItem.service_type=='pharmacy_item')")

File: oner-php/app/PharmacyItem.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class PharmacyItem extends Model 
{    
    protected $table = 'pharmacy_item';
    const CREATED_AT = 'created_time';
    const UPDATED_AT = 'updated_time';
    
    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'brand_name',
        'pharmacy_category_id',
        'universal_product_code',
        'generic_name',
        'form',
        'strength',
        'uom',
        'packaging',
        'created_user_id',
        'updated_user_id',
    ];

    public function pharmacy_category(){
        return $this->hasOne('App\PharmacyCategory','id','pharmacy_category_id');
    }

    public function pharmacy_unit_conversion(){
        return $this->hasOne('App\PharmacyUnitConversion','pharmacy_item_id','id');
    }

    public function pharmacy_inventory(){
        return $this->hasOne('App\PharmacyInventory','pharmacy_item_id','id');
    }

    public function service_item(){
        return $this->hasOne('App\ServiceItem','relation_id','id')->where('service_type','pharmacy_item');
    }

}
